packages/forms/src/model.ts

```diff
@@ -2112,23 +2116,31 @@ export class FormArray extends AbstractControl {
   /**
    * Remove the control at the given `index` in the array.
    *
-   * @param index Index in the array to remove the control
+   * @param index Index in the array to remove the control.  If `index` is negative, wraps around
+   *     from the back. If `index` is greatly negative (less than `-length`), removes the first
+   *     element. This behavior is the same as `Array.splice(index, 1)`.
    * @param options Specifies whether this FormArray instance should emit events after a
    *     control is removed.
    * * `emitEvent`: When true or not supplied (the default), both the `statusChanges` and
    * `valueChanges` observables emit events with the latest status and value when the control is
    * removed. When false, no events are emitted.
    */
   removeAt(index: number, options: {emitEvent?: boolean} = {}): void {
-    if (this.controls[index]) this.controls[index]._registerOnCollectionChange(() => {});
-    this.controls.splice(index, 1);
+    // Adjust the index, then clamp it at no less than 0.
+    let adjusted = this._adjustIndex(index);
```

Comment:
I'd propose adding the `Index` as well:
```suggestion
    let adjustedIndex = this._adjustIndex(index);
```

Reply:
Done.